spec/lang/types.md

```diff
@@ -197,18 +197,30 @@ And we also define how to compute the actual size and alignment.
 
 ```rust
 impl TupleHeadLayout {
+    /// The actual align of the tail, considering the packed attribute
+    fn capped_tail_align(self, tail_align: Align) -> Align {
+        match self.packed_align {
+            Some(packed_align) => tail_align.min(packed_align),
+            None => tail_align,
+        }
+    }
+
+    /// Where the tail starts, given the aligment of the tail type.
     pub fn tail_offset(self, tail_align: Align) -> Offset {
-        // TODO: support packed self
-        let tail_offset = Size::from_bytes(self.end.bytes().next_multiple_of(tail_align.bytes())).unwrap();
+        let capped_tail_align = self.capped_tail_align(tail_align);
+        // `self.end.align_to(capped_tail_align)`.
+        let tail_offset = Size::from_bytes(self.end.bytes().next_multiple_of(capped_tail_align.bytes())).unwrap();
```

Comment:
Might be worth actually adding `align_to` to our `Align` type?

Reply:
To the `Size` type, right?

Reply:
Ah right. With an `Align` argument.